woocommerce/woocommerce
# Add WooCommerce Extension environment scaffold

### All Submissions:






### Changes proposed in this Pull Request:



Add a one CLI line command to scaffold a WooCommerce development environment. The new package is to be published on NPM, but until that happens a local reference can also be used.

I'm building on top of [Gutenberg's Create Block](https://github.com/WordPress/gutenberg/tree/2a503082a1933989c04d336dd3aec7f32fb32d57/packages/create-block) which is primarily used to scaffold new blocks, but it can also be used for [project templates](https://github.com/WordPress/gutenberg/blob/2a503082a1933989c04d336dd3aec7f32fb32d57/packages/create-block/docs/external-template.md). This PR aims to make an initial stab at validating the idea as well as use of Create Block. Notable improvements to the output will be:
* Improve PHP folder structure
* Include example usage of `wc_get_container`
* Include better examples of wordpress and/or woocommerce components
* Include notices if WooCommerce isn't available or not loaded in time
* Include notices if build commands or composer install haven't been run
* Improve the PHP unit test to something more relevant
* Improve resulting README
* Include JS unit test

Some upstream pull requests to Create Block may be needed:
* Remove `block.json` from the output
* Dynamically provide `.wp-env` file so we can link to latest instead of hard coding `https://downloads.wordpress.org/plugin/woocommerce.7.1.0.zip`








### How to test the changes in this Pull Request:

1. Pull down this branch locally
2. `cd` to a new directory and run the scaffold command
```
npx @wordpress/create-block -t ../path/to/woocommerce/packages/js/create-extension my-extension-name
```
3. Install and build your new extension
```
cd my-extension-name
npm install
composer install
npm run build
``` 
4. `wp-env start` and navigate to your new extension's page, WooCommerce > my-extension-name
5. Take a look around the files, make sure everything makes sense







### Other information:




### FOR PR REVIEWER ONLY:
